src/mds/PurgeQueue.cc

```diff
@@ -588,8 +588,14 @@ void PurgeQueue::_execute_item(
   gather.set_finisher(new C_OnFinisher(
                       new LambdaContext([this, expire_to](int r){
     std::lock_guard l(lock);
-    _execute_item_complete(expire_to);
 
+    if (r == -EBLACKLISTED) {
+      finisher.queue(on_error, r);
+      on_error = nullptr;
+      return;
+    }
+
+    _execute_item_complete(expire_to);
```

Comment:
Also need to update Filer::_do_purge_range(). It ignores all IO errors

Reply:
Done.